python/dlisio/ext/core.cpp

```diff
@@ -392,11 +392,10 @@ noexcept (false) {
     assert(std::string(pre_fmt) == "");
     assert(std::string(post_fmt) == "");
 
-    dl::record record;
```

Comment:
Previously record data was reserved here. Now you reserve data in c itself.
Can we somehow spoil any memory/ownership model?